src/backend/joanie/core/models/products.py

```diff
@@ -1144,6 +1144,12 @@ def set_installment_refunded(self, installment_id):
         Set the state of an installment to `refunded` in the payment schedule.
         """
         ActivityLog.create_payment_refunded_activity_log(self)
+
+        if self.state != enums.ORDER_STATE_REFUNDING:
+            raise ValidationError(
+                "Cannot set installment to refunded if order state in not `refunding`"
+            )
+
```

Comment:
IMO this is not a good idea as if someone refund a transaction from Payzen we will not be able to store this information...

Reply:
True, i've forgetten that there could be an action made directly from the backoffice of Payzen.?